src/DnsHelper.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Yiisoft\NetworkUtilities;
+
+class DnsHelper
+{
+    /**
+     * @param string $hostname hostname without dot at end
+     */
+    public static function checkMx(string $hostname): bool
+    {
+        $hostname .= '.';
+        try {
+            if (!@dns_check_record($hostname, 'MX')) {
+                return false;
+            }
+            $result = @dns_get_record($hostname, DNS_MX);
+            return $result !== false && count($result) > 0;
+        } catch (\Throwable $t) {
```

Comment:
If the intent is to ignore any exception then a comment indicating it should be added inside catch block.

Reply:
Same about other empty catch blocks.

Reply:
corrected